pymc3/distributions/simulator.py

```diff
@@ -115,10 +115,10 @@ def random(self, point=None, size=None):
         array
         """
         params = draw_values([*self.params], point=point, size=size)
-        if size is None:
+        if not size:
```

Comment:
```suggestion
        if len(size) != 0:
```

Reply:
Oh, I am checking for `size` being empty tuple here. So, do you mean `len(size) == 0`?

Reply:
Yes.

Reply:
But if there is a call to this random method directly, then `size` will be `None`. I prefer to leave it as is, since it handles both cases (being `None` or empty tuple).

Reply:
Relying on the truthiness of an object can lead to really subtle and difficult to debug bugs. In that case I would be explicit, even if it doesn't look as clean:
`if size is None or len(size) == 0:`

Reply:
> can lead to really subtle and difficult to debug bugs

[Indeed](https://github.com/pandas-dev/pandas/pull/34768)

Reply:
Yes. I like this idea. Now, I have opted to use `to_tuple` to better handle corner cases.